tests/integration/devfile/devfile_suite_test.go

```diff
@@ -3,9 +3,18 @@ package devfile
 import (
 	"testing"
 
+	. "github.com/onsi/ginkgo"
 	"github.com/openshift/odo/tests/helper"
 )
 
 func TestDevfiles(t *testing.T) {
 	helper.RunTestSpecs(t, "Devfile Suite")
 }
+
+var _ = BeforeSuite(func() {
+	const registryName string = "DefaultDevfileRegistry"
+	const addRegistryURL string = "https://registry.stage.devfile.io"
+
+	// Use staging OCI-based registry for tests to avoid a potential overload
+	helper.CmdShouldPass("odo", "registry", "update", registryName, addRegistryURL)
```

Comment:
@kadel Trying to understand if this would be a good idea,
On one side we would detect errors early before the registry changes land into production?
but then anytime a small incompatible change is pushed to stage, it would break odo's test.

Reply:
@girishramnani Copying the second bullet point of the associated issue here: 

- The staging and production registries are largely identical, the only difference is when they are updated

    - Staging is updated with each commit to master in devfile/registry, production is updated manually each Wednesday
   
IMHO, we don't expect a big difference between staging vs. production registries. Early detection of a potential break would give more benefit than finding an error on the production registry that requires fixing anyway.

Reply:
I totally agree with @josephca.

> but then anytime a small incompatible change is pushed to stage, it would break odo's test.

This is exactly the reason why we should test against staging.  To get an early warning. 
If breaking change gets merged into "stage" we want to know about it before it gets into "prod" and fix it before it gets to "prod"

Reply:
Cool, merging